File: frontend/src/api/voice.ts
import { api } from './client'

export interface VoiceResult {
  intent: string; entities: Record<string, unknown>; action_taken: string
  response_message: string; log_id?: string; reminder_id?: string; success: boolean
}

export const voiceApi = {
  interpret: (transcript: string, baby_id: string) =>
    api.post<VoiceResult>('/voice/interpret', { transcript, baby_id }).then(r => r.data),
}

File: backend/app/schemas/voice.py
from pydantic import BaseModel


class VoiceInterpretIn(BaseModel):
    transcript: str
    baby_id: str


class VoiceResultOut(BaseModel):
    intent: str
    entities: dict
    action_taken: str
    response_message: str
    log_id: str | None = None
    reminder_id: str | None = None
    success: bool
